curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,530 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
+When using recursion, which statement is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+The function always runs infinitely.
 
 ---
 
-Placeholder distractor 2
+The function uses a while loop.
 
 ---
 
-Placeholder distractor 3
+The function has no parameters.
 
 #### --answer--
 
-Placeholder answer
+The function calls itself.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this return?
+
+```javascript
+function join(arr) {
+    if (arr.length <= 1) return arr;
+    return [arr[0]].concat(join(arr.slice(1)));
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+A reversed array.
 
 ---
 
-Placeholder distractor 2
+The first element only.
 
 ---
 
-Placeholder distractor 3
+An empty array.
 
 #### --answer--
 
-Placeholder answer
+A copy of the original array.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What technique can improve recursive performance?
 
 #### --distractors--
 
-Placeholder distractor 1
+Using more base cases.
 
 ---
 
-Placeholder distractor 2
+Adding more parameters.
 
 ---
 
-Placeholder distractor 3
+Using global variables.
 
 #### --answer--
 
-Placeholder answer
+Memoization
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this recursive function do?
+
+```javascript
+function repeat(str) {
+    return str + repeat(str);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Return the string twice.
 
 ---
 
-Placeholder distractor 2
+Return undefined.
 
 ---
 
-Placeholder distractor 3
+Create an empty string.
 
 #### --answer--
 
-Placeholder answer
+Cause a stack overflow.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What HTML structure is typically traversed recursively?
 
 #### --distractors--
 
-Placeholder distractor 1
+Style sheets
 
 ---
 
-Placeholder distractor 2
+Meta tags
 
 ---
 
-Placeholder distractor 3
+Script tags
 
 #### --answer--
 
-Placeholder answer
+The DOM tree
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In JavaScript, what keyword can help prevent infinite recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+`loop`
 
 ---
 
-Placeholder distractor 2
+`console`
 
 ---
 
-Placeholder distractor 3
+`try`
 
 #### --answer--
 
-Placeholder answer
+`if`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does this recursive function do?
+
+```javascript
+function reverseString(str) {
+  if (str === "") return "";
+  return reverseString(str.slice(1)) + str[0];
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Duplicates a string
 
 ---
 
-Placeholder distractor 2
+Removes vowels
 
 ---
 
-Placeholder distractor 3
+Removes spaces
 
 #### --answer--
 
-Placeholder answer
+Reverses a string.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the following code output?
+
+```javascript
+function isEven(n) {
+  if (n === 0) return true;
+  if (n === 1) return false;
+  return isEven(n - 2);
+}
+console.log(isEven(4));
+```
```

Comment:
maybe this is not the best example? it's a quite resource intesive way to check if a number is even or not

Reply:
Yeah that one wasn't my best 😂

Do you want my to make it just an `if/else` or should I just swap it out for a modulus `%`?

Thanks buddy 😄

Reply:
I changed this one to:

```javascript
function isEven(n) {
  return n === 0 ? true : n === 1 ? false : isEven(n - 2);
}
console.log(isEven(4));
```

Do you think that is too complex, or would you like me to go in another direction?
Thanks